File: problems/src/test/java/ReverseIntegerTest.java
import org.junit.Test;
import testing.Sample;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class ReverseIntegerTest extends SampledTest {
    @Override
    public String getFileName() {
        return "reverse_integer.json";
    }

    @Test
    public void test() {
        for (TypedSample sample : getSamples(TypedSample[].class)) {
            for (ReverseInteger solution : ReverseInteger.values()) {
                assertWithMessage(sample.getMessage())
                    .that(solution.reverse(sample.input.x))
                    .isEqualTo(sample.output);
            }
        }
    }

    public static class TypedSample extends Sample<TypedSample.Input, Integer> {
        public static class Input {
            public int x;
        }
    }
}

File: typescript/__tests__/problems1_100/reverse-integer.test.ts
import {strictEqual} from 'assert';
import SOLUTIONS from '../../src/problems1_100/reverse-integer';
import getSamples from '../../../testing/js/src/sample/samples';

type Sample = {
  input: {
    x: number,
  },
  output: number,
  message: string,
}

describe(
    'ReverseInteger',
    () =>
        test(
            'test',
            () =>
                SOLUTIONS.forEach(solution =>
                    getSamples().forEach((sample: Sample) =>
                        strictEqual(
                            solution.reverse(sample.input.x),
                            sample.output,
                            sample.message,
                        ),
                    ),
                ),
        ),
);
